src/vs/workbench/contrib/preferences/browser/settingsTreeModels.ts

```diff
@@ -291,30 +292,22 @@ export class SettingsTreeSettingElement extends SettingsTreeElement {
 	}
 
 	matchesAnyFeature(featureFilters?: Set<string>): boolean {
-		const features = tocData.children!.filter(child => child.id === 'features')[0].children!.map(feature => {
-			return {
-				name: feature.id.substring(9),
-				values: feature.settings!.map(setting => setting.toString().substring(0, setting.toString().length - 2))
-			};
-		});
 
 		if (!featureFilters || !featureFilters.size) {
 			return true;
 		}
 
-		const result = Array.from(featureFilters).map(filter => {
-			const foundFeature = features.find(item => item.name === filter);
-			if (foundFeature) {
-				if (foundFeature.values.length === 1) {
-					return Array.from(featureFilters).some((feature: string) => this.setting.key.toLowerCase().startsWith(feature) && !this.setting.extensionInfo);
-				} else {
-					return Array.from(featureFilters).some((feature: string) => (this.setting.key.toLowerCase().startsWith(feature) || this.setting.key.toLowerCase().startsWith(foundFeature.values[1])) && !this.setting.extensionInfo);
-				}
+		const features = tocData.children!.find(child => child.id === 'features')!.children;
```

Comment:
What if I write `@feature:blah`, or if we remove a feature in the future? This should not throw if it can't find the feature